### PR TITLE
Move to app deployment

### PR DESCRIPTION
**Pull request by MarcelMue**

### CODE REVIEW

File: prow/configs.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ github_reporter:
 deck:
   external_agent_logs:
     - agent: tekton-pipeline
-      url_template: 'http://log-aggregator:8000/logs/{{.ObjectMeta.Name}}'
+      url_template: {{`'http://log-aggregator:8000/logs/{{.ObjectMeta.Name}}'`}}
```
Comment (MarcelMue) [inline prow/configs.yaml:16]: This was done to basically render correctly again. `helm` was trying to render the `ObjectMeta.Name` field.

File: prow/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+configMapGenerator:
```
Comment (MarcelMue) [inline prow/kustomization.yaml:3]: This fills the configmaps with the actual config from files.

File: tekton/tekton.yaml [hunk 3]
```diff
@@ -0,0 +1,1202 @@
+# Copyright 2019 The Tekton Authors
```
Comment (MarcelMue) [inline tekton/tekton.yaml:1]: This is taken straight from upstream tekton - this way we can apply this together.

**Comment by MarcelMue** [comment]
This is not working except for the `ingress` issues with `deck` and `hook`.

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
+version: 2.1
+
+orbs:
+  architect: giantswarm/architect@0.11.0
+
+jobs:
+  build:
+    docker:
+      - image: quay.io/giantswarm/architect
```
Comment (stone-z) [inline .circleci/config.yml:9]: Reminder: version here

File: secrets/gs-token.yaml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
```
Comment (MarcelMue) [inline secrets/gs-token.yaml:1]: These are just here to reference the secrets we have - this way we don't have to remember the namespace/structure.

**Comment by MarcelMue** [comment]
@giantswarm/team-ludacris PTAL - I want to merge here soon. Already a lot of changes and having smaller PRs would be nice :D

File: README.md [hunk 6]
```diff
@@ -91,9 +91,9 @@ make configs
 
 Generated secrets are stored in the LastPass of our organization.
 
-- Access credentials for [Tityos Bot](https://github.com/tityosbot)
-- Github OAuth Secret
-- Random Github webhooks token
+- Access credentials for [Tityos Bot](https://github.com/tityosbot) (secret named `hmac-token` in `prow` namespace)
+- Github OAuth Secret (secret named `oauth-token` in `prow` namespace)
+- Random Github webhooks token (secret names `webhook-certs` in `tekton-pipelines` namespace)
```
Comment (tfussell) [inline README.md:96]:
```suggestion
- Random Github webhooks token (secret named `webhook-certs` in `tekton-pipelines` namespace)
```

Comment (tfussell) [inline README.md:96]: Why is this "random"?

File: crds/tekton-dashboard.yaml [hunk 7]
```diff
@@ -0,0 +1,44 @@
+apiVersion: apiextensions.k8s.io/v1beta1
```
Comment (tfussell) [inline crds/tekton-dashboard.yaml:1]: These are applied outside of helm I guess?

File: tekton/dashboard/kustomization.yaml [hunk 8]
```diff
@@ -0,0 +1,4 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+- tekton-dashboard-release-readonly.yaml
```
Comment (tfussell) [inline tekton/dashboard/kustomization.yaml:4]: Why is this named `readonly`?

File: README.md (hunk 6 above)
Comment (MarcelMue) [reply README.md]: I am not sure why the components need it. I will give the README a full re-write in a separate PR.

File: crds/tekton-dashboard.yaml (hunk 7 above)
Comment (MarcelMue) [reply crds/tekton-dashboard.yaml:1]: Yes - we would still have to `kubectl apply` these. I am not sure what a nice way to handle them would be.

File: tekton/dashboard/kustomization.yaml (hunk 8 above)
Comment (MarcelMue) [reply tekton/dashboard/kustomization.yaml:4]: It was readonly before - uncertain where this name came from.

File: crds/tekton-dashboard.yaml (hunk 7 above)
Comment (tfussell) [reply crds/tekton-dashboard.yaml:1]: Shouldn't be needed often anyways.

**Comment by MarcelMue** [comment]
Going with this - feel free to still leave comments. I will continue with smaller PRs today.

File: tekton/dashboard/kustomization.yaml (hunk 8 above)
Comment (stone-z) [reply tekton/dashboard/kustomization.yaml:4]: It is `readonly` from upstream - there is an alternative Tekton dashboard which lets you create/modify/delete Tekton resources directly

**Pull request merged**